experiments/download_ssl4eo.py

```diff
@@ -396,23 +393,20 @@ def get_random_patches_grid(
     # avoid strong overlap
     try:
         new_coord = (coords[0], coords[1])
-        gridIndex = (math.floor(new_coord[0] + 180), math.floor(new_coord[1] + 90))
-
-        if gridIndex not in grid_dict.keys():
-            grid_dict[gridIndex] = {new_coord}
-        else:
-            for coord in grid_dict[gridIndex]:
-                distance = np.sqrt(
-                    sampler.deg2km(abs(new_coord[0] - coord[0])) ** 2
-                    + sampler.deg2km(abs(new_coord[1] - coord[1])) ** 2
-                )
-                if distance < (1.5 * radius / 1000):
-                    raise OverlapError
-            grid_dict[gridIndex].add(new_coord)
+        for i in rtree_obj.nearest(new_coord, num_results=1, objects=True):
+            distance = np.sqrt(
+                sampler.deg2km(abs(new_coord[0] - i.bbox[2])) ** 2
+                + sampler.deg2km(abs(new_coord[1] - i.bbox[3])) ** 2
+            )
+            if distance < (1.5 * radius / 1000):
+                raise OverlapError
+        rtree_obj.insert(
+            len(rtree_obj) - 1, (new_coord[0], new_coord[1], new_coord[0], new_coord[1])
```

Comment:
This computes the size of the R-tree every single time, which is probably slow. We can either use a counter variable or just use 0 for all indices (we aren't using the index number anyway).